spring-kafka/src/test/java/org/springframework/kafka/core/KafkaTemplateTests.java

```diff
@@ -265,6 +267,29 @@ public boolean isInterestedInSuccess() {
 	}
 
 	@Test
+	public void withCallback() throws Exception {
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		DefaultKafkaProducerFactory<Integer, String> pf = new DefaultKafkaProducerFactory<>(senderProps);
+		KafkaTemplate<Integer, String> template = new KafkaTemplate<>(pf);
+		template.setDefaultTopic(INT_KEY_TOPIC);
+		final CountDownLatch latch = new CountDownLatch(1);
+		template.send(new ProducerRecord<Integer, String>(INT_KEY_TOPIC, "foo"), new Callback() {
+
+			@Override
+			public void onCompletion(RecordMetadata metadata, Exception exception) {
+				latch.countDown();
+			}
+
+		});
+		template.flush();
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+
+		//Drain the topic
+		KafkaTestUtils.getSingleRecord(consumer, INT_KEY_TOPIC);
+		pf.destroy();
+	}
+
+	@Test
 	public void testWithCallback() throws Exception {
```

Comment:
Well, my idea that we revert all the changes and present only the commit with the Docs polishing.

If we would have some comment in the doc before, you wouldn't came up with the current work, I think...

Otherwise, yes, we can just close this PR without any changes.